prefect_aws/ecs.py

```diff
@@ -0,0 +1,1019 @@
+    ```
+
+    Run a task and stream the container's output to the local terminal. Note an
+    execution role must be provided with permissions: logs:CreateLogStream,
+    logs:CreateLogGroup, and logs:PutLogEvents.
+    ```python
+    ECSTask(
+        command=["echo", "hello world"],
+        stream_output=True,
+        execution_role_arn="..."
+    )
+    ```
+
+    Run a task using an existing task definition as a base
+    ```python
+    ECSTask(command=["echo", "hello world"], task_definition_arn="arn:aws:ecs:...")
+    ```
+
+    Run a task with a specific image
+    ```python
+    ECSTask(command=["echo", "hello world"], image="alpine:latest")
+    ```
+
+    Run a task with custom memory and CPU requirements
+    ```python
+    ECSTask(command=["echo", "hello world"], memory=4096, cpu=2048)
+    ```
+
+    Run a task with custom environment variables
+    ```python
+    ECSTask(command=["echo", "hello $PLANET"], env={"PLANET": "earth"})
+    ```
+
+    Run a task in a specific ECS cluster
+    ```python
+    ECSTask(command=["echo", "hello world"], cluster="my-cluster-name")
+    ```
+"""
+import copy
+import sys
+import time
+import warnings
+from typing import Any, Dict, Generator, List, Optional, Tuple
+
+import boto3
+import yaml
+from anyio.abc import TaskStatus
+from prefect.docker import get_prefect_image_name
+from prefect.infrastructure.base import Infrastructure, InfrastructureResult
+from prefect.utilities.asyncutils import run_sync_in_worker_thread, sync_compatible
+from pydantic import Field, root_validator
+from typing_extensions import Literal
+
+from prefect_aws import AwsCredentials
+
+# Internal type alias for ECS clients which are generated dynamically in botocore
+_ECSClient = Any
+
+
+class ECSTaskResult(InfrastructureResult):
+    """The result of a run of an ECS task"""
+
+
+PREFECT_ECS_CONTAINER_NAME = "prefect"
+ECS_DEFAULT_CPU = 1024
+ECS_DEFAULT_MEMORY = 2048
+
+
+def get_prefect_container(containers: List[dict]) -> Optional[dict]:
+    """
+    Extract the Prefect container from a list of containers or container definitions
+    If not found, `None` is returned.
+    """
+    return get_container(containers, PREFECT_ECS_CONTAINER_NAME)
+
+
+def get_container(containers: List[dict], name: str) -> Optional[dict]:
+    """
+    Extract the a container from a list of containers or container definitions
+    If not found, `None` is returned.
+    """
+    for container in containers:
+        if container.get("name") == name:
+            return container
+    return None
+
+
+class ECSTask(Infrastructure):
+    """
+    <span class="badge-api experimental"/>
+
+    Run a command as an ECS task.
+
+    Note this block is experimental. The interface may change without notice.
+    """
+
+    _block_type_slug = "ecs-task"
+    _block_type_name = "ECS Task"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/1jbV4lceHOjGgunX15lUwT/db88e184d727f721575aeb054a37e277/aws.png?h=250"  # noqa
```

Comment:
This should fix the HTML showing up in the blocks UI.
```suggestion
    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/1jbV4lceHOjGgunX15lUwT/db88e184d727f721575aeb054a37e277/aws.png?h=250"  # noqa
    _description = "Run a command as an ECS task. Note this block is experimental. The interface may change without notice."
```

Reply:
Can confirm that this resolves the raw HTML displaying in the blocks UI
![Screen Shot 2022-09-07 at 11 57 28 AM](https://user-images.githubusercontent.com/12350579/188936774-3db24b85-35c5-4ec8-8e50-d184bab532e3.png)

Reply:
👍 thanks!